docs/en/02_Developer_Guides/00_Model/10_Versioning.md

```diff
@@ -318,6 +329,47 @@ The ownership relationship is tracked through an `[image]` [shortcode](/develope
 which is automatically transformed into an `<img>` tag at render time. In addition to storing the image path,
 the shortcode references the database identifier of the `Image` object.
 
+### Changesets, a.k.a "Campaigns"
+
+Changes to many DataObjects can grouped together using the `ChangeSet` object, better known by its frontend name, "Campaign" (provided the `campaign-admin` module is installed). By grouping a series of content changes together as one cohesive unit, content editors can bulk publish or revert an entire body of content all at once, which affords them much more power and control over interdependent content types.
+
+Records can be added to a changeset in the CMS by using the "Add to campaign" button
+that is available on the edit forms of all pages and files. Programmatically, this is done by creating a `SilverStripe\Versioned\ChangeSet` object and invoking its `addObject(DataObject $record)` method.
+
+<div class="info" markdown="1">
+Any DataObject can exist in any number of changesets, and even added to a changeset in advance of being published. While a record need not have modifications to be part of a changeset, for practical purposes, changesets are only concerned with records that have modifications.
+</div>
+
+By including an object in a changeset, the user is implicitly including all of that object's owned records (as declared by the `$owns` setting). This ensures that when a changeset is published (or reverted), the action cascades through not only all of the items explicitly added to the changeset, but also all of the records that each of those items owns.
+
+#### ChangeSet actions
+
+Actions available on the frontend, i.e. those which are intended to be triggered by an end user, include:
+
+* `$myChangeSet->addObject(DataObject $record)`: Add a record and all of its owned records to the changeset (`canEdit()` dependent)
+* `$myChangeSet->removeObject(DataObject $record)`: Removes a record and all of its owned records from the changeset (`canEdit()` dependent)
+* `$myChangeSet->::publish()`: Publishes all items in the changeset that have modifications, along with all their owned records (`canPublish()` dependent). Closes the changeset on completion.
+* `$myChangeSet->::revert()`: Reverts all items in the changeset that have modifications, along with all their owned records (`canRevert()` dependent). Closes the changeset on completion.
+ 
+ <div class="info" markdown="1">
+ Changesets cannot be "unpublished." Instead, a new changeset should be created to reverse the actions of the prior one.
```

Comment:
It's not really clear at the moment how this should be done; We should err in favour of not documenting behaviour we have no recommendation for at this stage. At 5.0 we may have a proper campaign history / revert UI and API, at which stage we should describe that behaviour instead.